packages/types/src/utils/executionAddress.ts

```diff
@@ -0,0 +1,48 @@
+import {keccak256} from "ethereum-cryptography/keccak.js";
+import {ByteVectorType} from "@chainsafe/ssz";
+
+export type ByteVector = Uint8Array;
+
+export class ExecutionAddressType extends ByteVectorType {
+  constructor() {
+    super(20, {typeName: "ExecutionAddress"});
+  }
+  toJson(value: ByteVector): unknown {
+    const str = super.toJson(value) as string;
+    return toChecksumAddress(str);
+  }
+}
+
+function isAddressValid(address: string): boolean {
+  return /^(0x)?[0-9a-f]{40}$/i.test(address);
+}
+
+/**
+ * Formats an address according to [ERC55](https://eips.ethereum.org/EIPS/eip-55)
+ */
+export function toChecksumAddress(address: string): string {
+  if (!isAddressValid(address)) {
+    throw Error(`Invalid address: ${address}`);
+  }
+
+  const rawAddress = address.toLowerCase().startsWith("0x") ? address.slice(2) : address;
```

Comment:
rawAddress is not actually converted to lower case, needs to be reassigned as strings are not mutated

Reply:
Must have slept in latest refactoring.. added test cases to catch those.